### PR TITLE
pj-rehearse: Add periodic jobs to the rehearsals 

### PR DESCRIPTION
**Pull request by droslean**
In this PR we enable the periodic jobs to the rehearse tool.

- [X] Adding periodic jobs to the job configuration.
- [X] Include periodic jobs to the execution
- [X] Update tests
- [X] Include periodic jobs to metrics
- [X] Update integration tests

### CODE REVIEW

File: pkg/rehearse/metrics.go [hunk 1]
```diff
@@ -84,8 +86,14 @@ func (m *Metrics) RecordChangedPresubmits(presubmits config.Presubmits) {
...
 
-func (m *Metrics) RecordOpportunity(toRehearse config.Presubmits, reason string) {
-	for _, jobs := range toRehearse {
+func (m *Metrics) RecordChangedPeriodics(periodics []prowconfig.Periodic) {
+	for _, job := range periodics {
+		m.ChangedPeriodics = append(m.ChangedPeriodics, job.Name)
+	}
+}
+
+func (m *Metrics) RecordOpportunity(presubmits config.Presubmits, periodics []prowconfig.Periodic, reason string) {
```
Comment (stevekuznetsov) [inline pkg/rehearse/metrics.go]: This is a smell -- if you only ever call it with presubmits or periodics lets just have two methods

Comment (droslean) [reply pkg/rehearse/metrics.go]: addressed

File: pkg/rehearse/metrics.go [hunk 2]
```diff
@@ -96,8 +104,22 @@ func (m *Metrics) RecordOpportunity(toRehearse config.Presubmits, reason string)
...
+	for _, job := range periodics {
+		if _, ok := m.Opportunities[job.Name]; !ok {
+			m.Opportunities[job.Name] = []string{reason}
+		} else {
+			m.Opportunities[job.Name] = append(m.Opportunities[job.Name], reason)
+		}
+	}
+}
+
+func (m *Metrics) RecordActual(presubmits []*prowconfig.Presubmit, periodics []prowconfig.Periodic) {
```
Comment (petr-muller) [inline pkg/rehearse/metrics.go]: ~~Should this method also be split to two, just like opportunities were?~~
Scratch that, I now see how this is used in `main` and keeping it as one makes sense.

File: pkg/rehearse/metrics.go [hunk 3]
```diff
@@ -96,8 +104,22 @@ func (m *Metrics) RecordOpportunity(toRehearse config.Presubmits, reason string)
...
-func (m *Metrics) RecordActual(rehearsals []*prowconfig.Presubmit) {
-	for _, job := range rehearsals {
+func (m *Metrics) RecordPeriodicsOpportunity(periodics []prowconfig.Periodic, reason string) {
+	for _, job := range periodics {
+		if _, ok := m.Opportunities[job.Name]; !ok {
+			m.Opportunities[job.Name] = []string{reason}
+		} else {
+			m.Opportunities[job.Name] = append(m.Opportunities[job.Name], reason)
+		}
+	}
```
Comment (bbguimaraes) [inline pkg/rehearse/metrics.go]: This (and the same above) can be just

```go
for _, job := range periodics {
    m.Opportunities[job.Name] = append(m.Opportunities[job.Name], reason)
}
```

File: cmd/pj-rehearse/main.go [hunk 4]
```diff
@@ -267,38 +267,45 @@ func rehearseMain() int {
...
+	metrics.RecordPresubmitsOpportunity(toRehearseClusterProfiles, "cluster-profile-change")
 	toRehearse.AddAll(toRehearseClusterProfiles)
 
 	jobConfigurer := rehearse.NewJobConfigurer(prConfig.CiOperator, prNumber, loggers, o.allowVolumes, changedTemplates, changedClusterProfiles, jobSpec.Refs)
 
 	presubmitsToRehearse := jobConfigurer.ConfigurePresubmitRehearsals(toRehearse)
-	metrics.RecordActual(presubmitsToRehearse)
 
-	if len(presubmitsToRehearse) == 0 {
+	changedPeriodics := diffs.GetChangedPeriodics(masterConfig.Prow, prConfig.Prow, logger)
```
[the remark this answers is not in the document]
Comment (droslean) [reply cmd/pj-rehearse/main.go]: addressed

File: pkg/rehearse/jobs_test.go [hunk 5]
```diff
@@ -49,36 +49,42 @@ func TestConfigurePresubmitRehearsals(t *testing.T) {
...
+			spec: v1.PodSpec{
 				Containers: []v1.Container{{}},
-				Volumes:    []v1.Volume{makeVoume(config.ClusterProfilePrefix + "changed-profile1")},
-			}),
+				Volumes:    []v1.Volume{makeVolume(config.ClusterProfilePrefix + "changed-profile1")},
+			},
+			expected: "rehearse-cluster-profile-changed-profile1-85c62707",
 		},
 	}
+
```
[the remark this answers is not in the document]
Comment (droslean) [reply pkg/rehearse/jobs_test.go]: Well, technically its the same thing. To test the changes in different test cases. I guess I we can just add more cases with multiple volumes will address your comment, right?

[the remark this answers is not in the document]
Comment (droslean) [reply pkg/rehearse/jobs_test.go]: Addressed

**Pull request merged**